### ISSUE TITLE
Tree: Nodes with long text are displaced

### ISSUE THREAD

**Issue by stonewoodman** [primefaces/primefaces#10361]
### Describe the bug

Nodes with long text are displaced
![image](https://github.com/primefaces/primefaces/assets/2945329/2bfb23da-3fbb-4bd4-9fd0-f9da53fb2bf2)

### Reproducer

```
        <h:form id="form">
            <div style="width: 500px">
                <p:tree id="testTree" value="#{indexController.root}" var="node">
                    <p:treeNode>
                        <h:outputText value="#{node}"/>
                    </p:treeNode>
                </p:tree>
            </div>
        </h:form>
```

```
@Named
@ViewScoped
public class IndexController implements Serializable
{
    private TreeNode<String> root;

    public void init() {
        root = new DefaultTreeNode<>();
        new DefaultTreeNode<>("Test 1", root);
        new DefaultTreeNode<>("Test 2 extra long text (Lorem ipsum dolor sit amet, consectetur adipisici elit)", root);
        new DefaultTreeNode<>("Test 3", root);
        TreeNode<String> test3 = new DefaultTreeNode<>("Test 4", root);
        new DefaultTreeNode<>("Test 5", root);

        TreeNode<String> test3_1 = new DefaultTreeNode<>("Child 3.1 extra long text (Lorem ipsum dolor sit amet, consectetur adipisici elit)", test3);

        new DefaultTreeNode<>("Child 3.1.1", test3_1);
        new DefaultTreeNode<>("Child 3.1.2 extra long text (Lorem ipsum dolor sit amet, consectetur adipisici elit)", test3_1);
        new DefaultTreeNode<>("Child 3.1.3", test3_1);
    }

    public TreeNode<String> getRoot()
    {
        return root;
    }

    public void setRoot(TreeNode<String> root)
    {
        this.root = root;
    }
}
```

### Expected behavior

It looks clean and not displaced
![image](https://github.com/primefaces/primefaces/assets/2945329/87984334-4418-4546-be92-a7fe3e8276ef)

Here is a possible bugfix code
```
.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content .ui-treenode-leaf-icon,
.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content .ui-chkbox,
.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content .ui-treenode-icon,
.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content .ui-tree-toggler {
    flex: none;
}

.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content .ui-treenode-label {
    flex-grow: 1;
}
```

### PrimeFaces edition

None

### PrimeFaces version

13.0.0, 14.0.0-SNAPSHOT

### JSF implementation

MyFaces

### JSF version

2.3

### Java version

17

**Comment by melloware** [comment]
Hmm can you try this just to see if it works?

```css
.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content {
    white-space: nowrap;
}
```

**Comment by melloware** [comment]
OK just checking!  I will give your fix a test.

**Comment by melloware** [comment]
OK you are right but this also does the trick so we don't need to name all the other components.

```css
.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content * {
    flex: none;
}

.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content .ui-treenode-label {
    flex-grow: 1;
}
```

**Comment by melloware** [comment]
actually I am not seeing the `flex-grow: 1` have any effect this just works for me?

```css
.ui-tree .ui-tree-container .ui-treenode .ui-treenode-content * {
    flex: none;
}
```

Does that work for you?

**Comment by stonewoodman** [comment]
yes, this works too and is shorter. perfect :)

**Comment by stonewoodman** [comment]
This bug is available in version 13.0.0 too
